### PR TITLE
docs: Add example for a simple compound component

### PR DESCRIPTION
**Pull request by vibdev**
## Summary

Whenever I want to use the compound component pattern it is hard to find a nice easy example to copy. Users often need to find a "simple" existing component, or dive into the documentation. This PR adds a super simple example of how to extend an existing CKR component with a customized wrapper. The goal is to have a one click place to get a skeleton that can be customized.

## Release Category
Documentation

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Areas for Feedback? (optional)

- [ ] Code
- [x] Documentation
- [ ] Testing
- [ ] Codemods

## Thank You!

a pilot saying: copy that

### CODE REVIEW

File: modules/react/_examples/stories/examples/compoundComponent/CustomCard.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    <Card depth={2} {...elemProps}>
+      {children}
+    </Card>
+  );
+});
+
+/**
+ * Usage
+ */
+export const Basic = () => {
```
Comment (alanbsmith) [inline modules/react/_examples/stories/examples/compoundComponent/CustomCard.tsx:107]:
```suggestion
export const Template = () => {
```

File: modules/react/_examples/stories/CompoundComponent.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import {Meta, Story} from '@storybook/addon-docs';
+import {Basic} from './examples/compoundComponent/CustomCard';
```
Comment (alanbsmith) [inline modules/react/_examples/stories/CompoundComponent.stories.mdx:2]:
```suggestion
import {Template} from './examples/compoundComponent/CustomCard';
```

File: modules/react/_examples/stories/examples/compoundComponent/CustomCard.tsx (hunk 1 above)
Comment (alanbsmith) [reply modules/react/_examples/stories/examples/compoundComponent/CustomCard.tsx:107]: Talked about this after standup. We appreciate you following the pattern and using `Basic` here, but it feels less like an example and more of a template.

**Review by mannycarrera4 — APPROVED** [review]

File: modules/react/_examples/stories/examples/compoundComponent/CustomCard.tsx (hunk 1 above)
Comment (vibdev) [reply modules/react/_examples/stories/examples/compoundComponent/CustomCard.tsx]: Done!

**Pull request merged**